modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1357,6 +1358,79 @@ impl LightningClientModule {
         }))
     }
 
+    /// Scan unspent incoming contracts for a payment hash that matches a
+    /// tweaked keys in the `indices` vector
+    pub async fn scan_receive_for_user_tweaked(
+        &self,
+        key_pair: KeyPair,
+        indices: Vec<u64>,
+    ) -> anyhow::Result<Amount> {
```

Comment:
Currently, this function can only be called only once the contract is funded, relying on the LNURL server to tell us about funded contracts. Later on we should migrate to a model where we scan for incoming offers and then start our normal state machines looking for contracts belonging to these offers.

Reply:
Even if we scan for offers instead of contracts, aren't we still relying on the LNURL server to tell us the tweak that it used for the public key? If it doesn't reveal the tweak and it is hard to guess by scanning (i.e not within some gap limit), I don't think the client will be able to create the `SecretKey` needed to claim the contract

Reply:
I misinterpreted this comment, discussed offline. Added `subscribe_ln_claim` in the short term to address this